### DISCUSSION TITLE
feature: Implement type guard in the isEnum() function

### DISCUSSION THREAD

**Discussion by kterui9019** [typestack/class-validator#2365] [Feature Ideas]
### Description
The `isEnum()` function does not implement type guard.
This is inconvenient when writing code like the following

```ts
const somefunction(value: string): USERTYPE => {
  if (!isEnum(value, USERTYPE)) throw new Error("not match enum");
  return value; // Type 'string' is not assignable to type 'USERTYPE'. ts(2322)
}

```

### Proposed solution
A type guard should be implemented in the isEnum() function.
I am now wrapping the `isEnum()` function in the type guard function myself.

```ts
function validateEnumValue<T>(value: any, enumeration: T): value is T[keyof T] {
  return isEnum(value, enumulation);
}
```

**Comment by kterui9019** [comment]
Specifically, I believe this can be accomplished by modifying the code as follows.
```ts
export function isEnum<T extends Record<string | number, unknown>>(value: unknown, entity: T): value is T[keyof T]  {
  const enumValues = Object.values(entity);
  return enumValues.includes(value);
}
```

I can send you a pull request if you like.

**Comment by braaar** [comment]
I'm moving this to the discussion tab, as we are trying to keep the issue count low. Please see [this issue](https://github.com/typestack/class-validator/issues/1775) for an explanation.
